### PR TITLE
Update Guideline

### PR DESCRIPTION
**Pull request by jsenon**
Add guideline/warning, that we need to have unique name in `port name` if we use multiple host and HTTPS protocol.

### CODE REVIEW

**Comment by geeknoid** [comment]
You'll need to cherry pick this into master.

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 1]
```diff
@@ -232,3 +232,49 @@ $ kubectl scale --replicas=0 deploy/istio-citadel -n istio-system
 {{< /text >}}
 
 This should stop Istio from restarting Envoy and disconnecting TCP connections.
+
+## Configure a multiple host gateway with SSL
+
+To configure the ingress gateway with TLS correctly for you to deploy multiple
+hosts, you must configure a unique port name with the following command:
```
Comment (frankbu) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: As I said in #2321, this doesn't describe the general problem. I suggest something like the following:

## Configuring multiple TLS hosts in a gateway

If you configure multiple gateway servers, either in a single `Gateway`, or spread across more than one that use the same `selector` labels, with the same port number and protocol HTTPS, then you must ensure that the corresponding port names are unique. For example:

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 2]
```diff
@@ -232,3 +232,51 @@ $ kubectl scale --replicas=0 deploy/istio-citadel -n istio-system
 {{< /text >}}
 
 This should stop Istio from restarting Envoy and disconnecting TCP connections.
+
+## Configuring multiple TLS hosts in a gateway
+
+If you configure multiple gateway servers, either in a single Gateway,
```
Comment (frankbu) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Put backticks around Gateway: `Gateway`

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 3]
```diff
@@ -232,3 +232,51 @@ $ kubectl scale --replicas=0 deploy/istio-citadel -n istio-system
...
+or spread across more than one that use the same selector labels,
```
Comment (frankbu) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Put backticks around selector: `selector`

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 4]
```diff
@@ -232,3 +232,51 @@ $ kubectl scale --replicas=0 deploy/istio-citadel -n istio-system
...
+    tls:
+      mode: SIMPLE
+      serverCertificate: /etc/istio/ingressgateway-bookinfo-certs/tls.crt
+      privateKey: /etc/istio/ingressgateway-bookinfo-certs/tls.key
+    hosts:
+    - "bookinfo.com"
+EOF
+{{< /text >}}
+
+If the port name is not unique, the `istio-pilot` pod returns the following error:
```
Comment (frankbu) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: If the port names are not unique, `istio-pilot` will return the following error:

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 5]
```diff
@@ -232,3 +232,51 @@ $ kubectl scale --replicas=0 deploy/istio-citadel -n istio-system
...
+
+{{< text plain >}}
+port https.443.HTTPS: non unique port name for HTTPS port
+{{< /text >}}
```
Comment (frankbu) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: where exactly is this? In the istio-pilot logs? Please explain where users would see it.

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 6]
```diff
@@ -232,3 +232,51 @@ $ kubectl scale --replicas=0 deploy/istio-citadel -n istio-system
...
+
+If you run a `curl` command, Istio returns the same error under the `SSL_SYSCALL_ERROR` value.
```
Comment (frankbu) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: Please explain exactly what happens. Does curl return 0 and print SSL_SYSCALL_ERROR ... ?

File: content/help/ops/traffic-management/deploy-guidelines/index.md [hunk 7]
```diff
@@ -232,3 +232,51 @@ $ kubectl scale --replicas=0 deploy/istio-citadel -n istio-system
...
+EOF
+{{< /text >}}
+
+If the port names are not unique, `istio-pilot` will return the following error in the log:
+
+{{< text plain >}}
+port https.443.HTTPS: non unique port name for HTTPS port
+{{< /text >}}
+
+If you run a `curl` command, command returns 0 and print `SSL_SYSCALL_ERROR`.
```
Comment (frankbu) [inline content/help/ops/traffic-management/deploy-guidelines/index.md]: s/command returns 0/it will return 0/

**Comment by frankbu** [comment]
Force merging because circleci error is unrelated.

**Comment by jsenon** [comment]
Thanks for merge @frankbu

**Pull request merged**